n2y/main.py

```diff
@@ -1,36 +1,100 @@
 import os
 import sys
 import argparse
+import re
 
 import yaml
+import pandoc
 
-from n2y import notion, simplify
+from n2y import converter, notion, simplify
 
 
 def main():
+    parser = argparse.ArgumentParser(description="Move data from Notion into YAML/markdown",
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("database", help="The Notion database id or share url")
+    parser.add_argument("--output", '-o',
+                        choices=["yaml", "markdown"], default="yaml",
+                        help=("Select output type\n"
+                              "  yaml - print yaml to stdout\n"
+                              "  markdown - create a mardown file per page"))
+    parser.add_argument("--image-path", help="Specify path where to save images")
+    parser.add_argument("--image-web-path", help="web path for images")
+    parser.add_argument("--plugins", help="plugin file")
+    parser.add_argument("--name-column", default='title',
+                        help=("Name of column containting the page name."
+                              "Lowercase letter and numbers. Replace spaces with underscore."))
+    args = parser.parse_args()
+
     ACCESS_TOKEN = os.environ.get("NOTION_ACCESS_TOKEN", None)
     if ACCESS_TOKEN is None:
         print("No NOTION_ACCESS_TOKEN environment variable is set", file=sys.stderr)
         return 1
 
-    parser = argparse.ArgumentParser(description="Move data from Notion into YAML files")
-    parser.add_argument("database", help="The Notion database id or share url")
-    parser.add_argument(
-        "--raw", action='store_true',
-        help="Dump the raw notion API data",
-    )
-    args = parser.parse_args()
     database_id = notion.id_from_share_link(args.database)
+
+    converter.IMAGE_PATH = args.image_path
+    converter.IMAGE_WEB_PATH = args.image_web_path
+    if args.plugins:
+        converter.load_plugins(args.plugins)
+
     client = notion.Client(ACCESS_TOKEN)
-    raw_rows = client.get_database(database_id)
-    if args.raw:
-        print(yaml.dump(raw_rows))
-        return 0
 
-    simplified_rows = simplify.flatten_database_rows(raw_rows)
-    print(yaml.dump(simplified_rows))
+    if args.output == 'markdown':
+        export_markdown(client, database_id, options=args)
+    elif args.output == 'yaml':
+        export_yaml(client, database_id, options=args)
+
     return 0
 
 
+def export_markdown(client, database_id, options):
+    raw_rows = client.get_database(database_id)
+
+    if options.name_column not in raw_rows[0]:
```

Comment:
I like that you have an argument that lets you select the name_column.

Instead of defaulting to the column with a name of "title", which may not exist, I think you can default to the column with the "title" data type. I believe this is guaranteed to exist and usually is title-like.

What happens now if there are